### PR TITLE
[doc] Add shape on creating a representation from another representation

### PR DESCRIPTION
**Pull request by gdaniel**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [x] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
+= Create a representation using the content of another representation as input
+
+== Context
+
+Representation event processors are isolated on purpose, but now we want to synchronize the state of a representation event processor with another one.
```
Comment (AxelRICHARD) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: `we want to` -> `we want to be able to`

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+In our example, both `FormEventProcessor` and `DiagramEventProcessor` could be refreshed simultaneously by having some specific parameter in `ChangeDescription`.
+
+The refresh policy can also be configured by an `IRepresentationRefreshPolicyProvider`, which would allow a specific representation to be refreshed on events that don't trigger a refresh on it usually (e.g. a new form representation could be refreshed by diagram events in addition to the regular semantic changes).
+
+=== Retrieve the current state of a representation
+
+The `FormDescription#getVariableManagerInitializer` API allows to initialize variables for a given form representation.
+
+We can leverage this form-only (for now) mechanism to retrieve the `IRepresentationEventProcessor` of the representation we want the state of:
```
Comment (AxelRICHARD) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: This sentence is difficult to read. May be a `,` a comma is missing somewhere, or may be a word is missing but I don't understand the `... the state of:` part at the end.

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                      .findFirst()
+              )
+              .orElse(null);
+      vm.put("editingContextEventProcessor", editingContextEventProcessor);
+      vm.put("representationEventProcessor", representationEventProcessor);
+      vm.put("representation", representationEventProcessor.getRepresentation());
+  }
+```
+
+The `representation` variable could the be used in the form to initialize its content (e.g. populate a tree widget).
```
Comment (AxelRICHARD) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: `could the` -> `could then`

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+=== Propagate changes on an existing representation
+
+The section above allows to create new representations that use the content of existing representations to intialize their content, but it may be necessary to apply mutations on the existing representation from actions in the new one (e.g. a form initialized with a diagram may have a button to change the visibility of elements on the diagram).
+In this case we want to make sure both representation stay synchronized.
+
+This can be done by leveraging the `IRepresentationEventProcessor#handle` method to perform an operation on another representation, but doing so requires to create a new ChangeDescription flux, which won't be known by the `EditingContextEventProcessor`.
+
+This means that the synchronization won't be complete: the operation will be executed, but the representation calling the operation won't be automatically updated. 
+
+Synchronization can be achieved done by manually calling `IRepresentationEventProcessor#refresh` with the local ChangeDescription flux on all the `IRepresentationEventProcessor` of the current `IEditingContextProcessor`:
```
Comment (AxelRICHARD) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: `can be achieved done by` -> `can be achieved by`

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc (hunk 1 above)
Comment (gdaniel) [reply doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: done

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc (hunk 2 above)
Comment (gdaniel) [reply doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: Indeed the sentence wasn't clear, I tried to rephrase it, let me know if it needs some more work

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc (hunk 3 above)
Comment (gdaniel) [reply doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: done

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc (hunk 4 above)
Comment (gdaniel) [reply doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: done

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 5]
```diff
@@ -0,0 +1,92 @@
+= Create a representation using the content of another representation as input
+
+== Context
+
+Representation event processors are isolated on purpose, but now we want to be able to synchronize the state of a representation event processor with another one.
```
Comment (pcdavid) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc:5]: It's a more limited use case that what is described here, but note that the _Representations_ view (in the right panel) supports displaying the structure/contents of a _Portal_ representation (see https://github.com/eclipse-sirius/sirius-web/pull/2897).

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+== Key Result
+
+We want:
+
+- To compute a representation in a representation event processor with the sate of another representation as input (e.g. a form using a diagram as input)
```
Comment (sbegaudeau) [inline doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: `sate` -> `state`

**Review by sbegaudeau — APPROVED** [review]
I'll rebase it and merge it but the content of the document is a bit odd.

On one side, shapes are made to describe the behavior we want to have and ADR are made to describe how we believe we implement do it. Here you have a shape talking almost exclusively about the technical details behind the feature you want to implement. And contrary to most ADR, it's a bit more a work in progress of how you found out you could to it.

That's not really an issue but in the end it feels a bit more like a "How to" document (see `/doc/how-to`) explaining how to create a representation using another representation as an input than either a shape or an ADR. It's still very interesting to have in the repository but we may move it / copy it there once it's fully done.

File: doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc (hunk 6 above)
Comment (sbegaudeau) [reply doc/iterations/2024.5/shapes/create_representation_using_content_of_another_representation.adoc]: I've fixed it

**Pull request merged**